deployment/README.md

```diff
@@ -0,0 +1,136 @@
+# CCNP Deployment Guide
+
+
+## Create TD
+
+CCNP is designed for collecting confidential computing primitives from confidential virtual machines. It's designed to run as daemon set on confidential virtual machines nodes, such as Intel Trust Domain (TD), in a Kubernetes cluster.
+You can use [cvm image rewriter](../tools/cvm-image-rewriter/README.md) to prepare a TD enlightened guest image and start a TD using [qemu-test.sh](../tools/cvm-image-rewriter/qemu-test.sh)
+
+## Prepare a K8S cluster with TD as worker nodes
+
+You can either create a K8S cluster in the TD or let the TD join an existing K8S cluster. Please choose one of the following step to make sure the K8S cluster is prepared with the TD running in it. CCNP will be deployed on the TD later.
+
+### Option 1: Create a K8S cluster on the TD
+After TDs are started, users need to setup a K8S cluster in the TDs. Please refer to the [k8s official documentation](https://kubernetes.io/docs/home/) for detailed steps. 
+
+_NOTE: If the cluster has only one node (master node), the taint on the node needs to be removed._
+
+### Option 2: Add the TD to an existing K8S cluster
+TBD
+
+## Deploy CCNP
+
+- [image-manager.sh](../deployment/script/image-manager.sh): The tool will build 4 images and push them to remote registry if required.
+- [deploy-ccnp.sh](../deployment/script/deploy-ccnp.sh): The tool will deploy CCNP services as DaemonSet on TDs in the K8S cluster.
+- [deploy-and-exec-ccnp-example.sh](../deployment/script/deploy-and-exec-ccnp-example.sh): The tool will deploy an example pod and show getting event logs, measurement and quote using CCNP in the pod.
+
+### Prerequisite
+- Install Helm on the TD nodes. Please refer to the [HELM quick start](https://helm.sh/docs/intro/quickstart/).
+- Install docker on the TD nodes. Please refer to [Get Docker](https://docs.docker.com/get-docker/)
+- Install python3-pip on the TD nodes. Please refer to [pip document](https://pip.pypa.io/en/stable/installation/)
+- Set access permission to TD device node on the TD nodes.
+```
+$ sudo mkdir -p /etc/udev/rules.d
+$ sudo touch /etc/udev/rules.d/90-tdx.rules
+
+# Check TD device node on TD
+$ ls /dev/tdx*
+
+# If above output is "/dev/tdx-guest"
+$ sudo bash -c 'echo "SUBSYSTEM==\"misc\",KERNEL==\"tdx-guest\",MODE=\"0666\"">/etc/udev/rules.d/90-tdx.rules'
+# If above output is "/dev/tdx_guest"
+$ sudo bash -c 'echo "SUBSYSTEM==\"misc\",KERNEL==\"tdx_guest\",MODE=\"0666\"">/etc/udev/rules.d/90-tdx.rules'
+
+$ sudo udevadm trigger
+$ sudo mkdir -p  /run/ccnp/uds
+$ sudo chmod o+w /run/ccnp/uds
+```
+
+### Build CCNP images
+
+Run below scripts to generate CCNP images. It will generate 5 images and push them to user specific registry.
```

Comment:
Will this also create the ccnp-example:0.1 image?

Reply:
It's ccnp-node-measurement-example